### PR TITLE
Fixes #63: Adds `scroll-to-focused-item` to enable scrolling to an item before focusing.

### LINKED ISSUE #63: Menu scrolls unexpectedly when selecting next/previous item
Fiddle: https://jsfiddle.net/ntw6trbk/
1. Focus any one of the 3 menus (tap "Item 0" or tap the dropdown trigger).
2. Press the down arrow 5 or so times.
#### Expected

Once the selection is at the bottom of the visible menu container, Down should scroll the menu down by one, so the selection is always at the bottom. (`select`)
#### Actual

Every 5 or so element, the menu scroll position jumps so the selected item is in the middle of the visible menu container instead of the bottom. (`paper-menu`, `paper-listbox`)

This is particularly disorienting when the list wraps, because it's not immediately obvious that it isn't just continuing the "move downward, scroll upward" behavior.

Also repros with the down arrow (selection should always be at the top, but the menu scrolls up occasionally).

### PR DESCRIPTION
**Pull request by bicknellr**

### CODE REVIEW

File: test/iron-menu-behavior.html [hunk 1]
```diff
@@ -379,6 +407,123 @@
             done();
           });
         });
+
+        suite('`scroll-to-focused-item`', function() {
```
Comment (cdata) [inline test/iron-menu-behavior.html:411]: Nice test suite 🍻

File: iron-menu-behavior.html [hunk 2]
```diff
@@ -197,6 +206,26 @@
...
         focusedItem.setAttribute('tabindex', '0');
+
+        // TODO(bicknellr): Most of this can be removed once the
+        // ScrollIntoViewOptions argument type gets general support.
+        // https://drafts.csswg.org/cssom-view/#dom-element-scrollintoview
+        if (this.scrollToFocusedItem) {
+          var selfRect = this.getBoundingClientRect();
+          var focusedItemRect = focusedItem.getBoundingClientRect();
+          var topOutside = focusedItemRect.top < selfRect.top;
+          var bottomOutside =
```
Comment (cdata) [inline iron-menu-behavior.html:217]: There is a middle-out joke here somewhere.

**Comment by cdata** [comment]
LGTM 👍

**Comment by bicknellr** [comment]
Just tried this with paper-dropdown-menu and it doesn't work because the iron-overlay is what ends up overflowing and scrolling, not the IronMenuBehavior thing inside. I wouldn't consider this broken with regards to IronMenuBehavior though but it needs another property to specify additional containers that might also crop out the item, or maybe it could recurse up the tree and apply this to all ancestors with scrollHeight > offsetHeight? Also, I'm not sure what the argument to `scrollIntoView` is meant to do if the element is outside it's nearest scrolling ancestor's scroll bounds in one direction and _that_ element is outside the scroll bounds of another ancestor in the opposite direction.

**Pull request closed without merging**